File: scripts/add-more-tools.js
const fs = require('fs');
const path = require('path');

const existingTools = JSON.parse(fs.readFileSync(path.join(__dirname, '..', 'data', 'tools.json'), 'utf-8'));
const existingNames = new Set(existingTools.map(t => t.name.toLowerCase()));
const existingIds = new Set(existingTools.map(t => t.id));

let idCounter = Math.max(...existingIds) + 1;
const getNextId = () => idCounter++;

const generateRatingBreakdown = (baseScore = 4.5) => ({
  ease_of_use: { score: Math.min(5, Math.max(3, baseScore + (Math.random() - 0.5))), note: "User-friendly interface" },
  output_quality: { score: Math.min(5, Math.max(3, baseScore + (Math.random() - 0.3))), note: "High-quality output" },
  features: { score: Math.min(5, Math.max(3, baseScore + (Math.random() - 0.4))), note: "Rich feature set" },
  value_for_money: { score: Math.min(5, Math.max(3, baseScore + (Math.random() - 0.2))), note: "Good value" },
  stability: { score: Math.min(5, Math.max(3, baseScore + (Math.random() - 0.3))), note: "Reliable performance" },
  support: { score: Math.min(5, Math.max(3, baseScore + (Math.random() - 0.5))), note: "Helpful support" }
});

const createTool = (name, category, pricing, description, descriptionEn, needsVpn = false, skillLevel = "beginner", bestFor = []) => {
  if (existingNames.has(name.toLowerCase())) return null;
  
  const baseScore = 4.0 + Math.random() * 0.8;
  
  return {
    id: getNextId(),
    name,
    description,
    category,
    pricing,
    url: `https://${name.toLowerCase().replace(/\s+/g, '-')}.com`,
    affiliate_link: "",
    icon_url: "",
    examples: [],
    needs_vpn: needsVpn,
    languages: ["English"],
    description_en: descriptionEn,
    rating: Math.round(baseScore * 10) / 10,
    rating_count: Math.floor(Math.random() * 5000) + 100,
    rating_breakdown: generateRatingBreakdown(baseScore),
    last_updated: "2026-06-01",
    skill_level: skillLevel,
    best_for: bestFor.length > 0 ? bestFor : ["General Use", "Productivity", "Content Creation"]
  };
};

const newToolsData = [
  ["AutoShorts AI", "Video", "Freemium", "AI-powered platform that automatically generates short-form videos for TikTok, Reels, and YouTube Shorts from text or blog posts. Includes auto-captioning and trending audio suggestions.", "Auto-generate short videos for social media platforms.", false, "beginner", ["TikTok", "Reels", "Shorts"]],
  ["Repurpose.io", "Productivity", "Freemium", "Automatically convert your content into multiple formats for different platforms. Transform YouTube videos into clips, podcasts into blog posts, and more.", "Convert content across multiple platforms automatically.", false, "beginner", ["Content Repurposing", "Social Media", "Automation"]],
  ["Castmagic", "Audio", "Paid", "AI-powered podcast production tool that generates show notes, timestamps, social media clips, and full transcripts automatically.", "AI podcast production with automated content generation.", false, "intermediate", ["Podcasting", "Show Notes", "Automation"]],
  ["Fireflies AI", "Productivity", "Freemium", "AI meeting assistant that transcribes, summarizes, and analyzes your meetings. Integrates with Zoom, Google Meet, and Teams.", "AI meeting transcription and analysis.", false, "beginner", ["Meetings", "Transcription", "Notes"]],
  ["AdCreative AI", "Productivity", "Paid", "Generate high-converting ad creatives and copy with AI. Create multiple variations for A/B testing in seconds.", "AI-powered ad creative generation.", false, "intermediate", ["Advertising", "Marketing", "Conversion"]],
  ["Runway ML", "Video", "Freemium", "Advanced AI video editing and generation platform. Features include绿幕 removal, motion tracking, and generative tools.", "Advanced AI video editing and generation.", false, "intermediate", ["Video Editing", "AI Generation", "Creative"]],
  ["PlayHT", "Audio", "Freemium", "Realistic AI voice generation with emotional expression. Create natural-sounding voiceovers for videos, podcasts, and e-learning.", "Realistic AI voice generation with emotions.", false, "beginner", ["Voiceover", "TTS", "Audio"]],
  ["Jasper Art", "Image", "Paid", "AI image generation integrated with Jasper's content platform. Create visuals that match your brand style.", "AI image generation for brands.", false, "intermediate", ["Brand Images", "Marketing", "Creative"]],
  ["Cohere", "Productivity", "Freemium", "Enterprise AI platform for building LLM-powered applications. Embeddings, classification, and semantic search.", "Enterprise AI platform for applications.", false, "advanced", ["Enterprise", "LLM", "API"]],
  ["Tabnine", "Code", "Freemium", "AI code completion that respects your codebase patterns and style. Works with all major IDEs and languages.", "AI code completion for developers.", false, "beginner", ["Code Completion", "IDE", "Productivity"]],
  ["Phind", "Code", "Free", "AI search engine designed for developers. Get instant answers to programming questions with code examples.", "AI search engine for developers.", false, "beginner", ["Search", "Code Help", "Development"]],
  ["Loom AI", "Video", "Freemium", "AI-powered video messaging platform. Auto-generated titles, summaries, and chapter markers.", "AI video messaging and recording.", false, "beginner", ["Video Messages", "Async", "Communication"]],
  ["Tome AI", "Productivity", "Freemium", "AI-powered presentation and storytelling tool. Create visually stunning presentations from text prompts.", "AI presentation and storytelling tool.", false, "beginner", ["Presentations", "Storytelling", "Slides"]],
  ["Beautiful.ai", "Productivity", "Paid", "AI-powered presentation software that automatically designs slides. Focus on content while AI handles the design.", "AI-powered presentation design.", false, "beginner", ["Presentations", "Design", "Productivity"]],
  ["Zapier AI", "Productivity", "Paid", "AI-powered workflow automation. Connect apps and automate tasks with natural language descriptions.", "AI workflow automation with Zapier.", false, "intermediate", ["Automation", "Integrations", "Productivity"]],
  ["Make AI", "Productivity", "Freemium", "Visual automation platform with AI capabilities. Build complex workflows without code.", "Visual automation with AI.", false, "intermediate", ["Automation", "Workflows", "No Code"]],
  ["Notion AI", "Productivity", "Paid", "AI writing assistant integrated into Notion. Summarize, brainstorm, and write faster within your workspace.", "AI assistant within Notion.", false, "beginner", ["Writing", "Notes", "Workspace"]],
  ["Craft AI", "Productivity", "Freemium", "AI-powered note-taking app with beautiful design. Generate content and organize your thoughts.", "AI-powered note-taking.", false, "beginner", ["Note-taking", "Design", "Organization"]],
  ["Logseq AI", "Productivity", "Free", "Privacy-first knowledge management with AI assistance. Outliner and graph view with local storage.", "Privacy-first knowledge management.", false, "intermediate", ["Knowledge", "Privacy", "Outliner"]],
  ["Napkin AI", "Productivity", "Freemium", "AI-powered tool for creating visual content from text. Turn your ideas into infographics and graphics.", "Create visual content from text.", false, "beginner", ["Graphics", "Infographics", "Visual"]],
  ["Gamma AI", "Productivity", "Freemium", "AI-powered presentation and document creator. Generate slides, websites, and documents from a single prompt.", "AI presentation and document creator.", false, "beginner", ["Presentations", "Documents", "AI Generation"]],
  ["Kling AI", "Video", "Freemium", "Chinese AI video generation platform by Kuaishou. High-quality video generation from text and images.", "Chinese AI video generation platform.", true, "intermediate", ["Video Generation", "AI", "Text-to-Video"]],
  ["Pika AI", "Video", "Freemium", "AI video generation platform for creating and editing videos. User-friendly interface for content creators.", "User-friendly AI video generation.", false, "beginner", ["Video Creation", "AI", "Content"]],
  ["Hailuo AI", "Video", "Freemium", "AI-powered video creation from text. Developed by MiniMax for high-quality video generation.", "AI video generation by MiniMax.", true, "intermediate", ["Video Generation", "AI", "Text-to-Video"]],
  ["Luma AI", "Image", "Freemium", "AI-powered 3D capture and generation. Create stunning 3D models from photos with Neural Radiance Fields.", "AI 3D capture and generation.", false, "intermediate", ["3D", "Capture", "Innovation"]],
  ["Meshy AI", "Image", "Freemium", "AI-powered 3D model creation from text or images. Transform concepts into 3D assets quickly.", "AI 3D model creation.", false, "intermediate", ["3D Models", "AI", "Design"]],
  [" Spline AI", "Productivity", "Freemium", "AI-assisted 3D design tool in browser. Create 3D scenes and animations with AI assistance.", "AI-assisted 3D design.", false, "intermediate", ["3D Design", "Browser", "Animation"]],
  ["RAGFlow", "Productivity", "Free", "Open-source RAG engine for enterprise. Deep document understanding with intelligent retrieval.", "Open-source RAG engine for documents.", false, "advanced", ["RAG", "Enterprise", "Documents"]],
  ["Dify", "Productivity", "Freemium", "Open-source LLM app development platform. Build AI applications with visual workflow.", "Open-source LLM app platform.", false, "intermediate", ["LLM Apps", "Development", "No Code"]],
  ["Flowise AI", "Productivity", "Free", "Drag-and-drop UI for building LLM flows. Create custom AI workflows without coding.", "Drag-and-drop LLM flows builder.", false, "intermediate", ["LLM Flows", "No Code", "Automation"]],
  ["AutoGen Studio", "Productivity", "Free", "Microsoft's open-source tool for building LLM agents. Create multi-agent workflows visually.", "Microsoft's LLM agent builder.", false, "advanced", ["LLM Agents", "Microsoft", "Automation"]],
  ["EmbedAI", "Productivity", "Freemium", "Create AI chatbots trained on your data. Build customer support and knowledge base bots easily.", "AI chatbots from your data.", false, "beginner", ["Chatbots", "Support", "Knowledge Base"]],
  ["Chatbase", "Productivity", "Freemium", "Build AI chatbots from your documents and websites. No coding required.", "Build chatbots from documents.", false, "beginner", ["Chatbots", "No Code", "Support"]],
  ["Voiceflow", "Productivity", "Freemium", "Design and deploy AI assistants across channels. Visual conversation designer.", "Visual AI assistant designer.", false, "intermediate", ["Voice Assistants", "Design", "Multi-channel"]]
];

const newTools = newToolsData.map(args => createTool(...args)).filter(Boolean);
const updatedTools = [...existingTools, ...newTools];

fs.writeFileSync(
  path.join(__dirname, '..', 'data', 'tools.json'),
  JSON.stringify(updatedTools, null, 2),
  'utf-8'
);

console.log(`✅ Added ${newTools.length} new tools`);
console.log(`📊 Total tools now: ${updatedTools.length}`);

File: scripts/add-more-tools.py
#!/usr/bin/env python3
"""Add 25 more unique AI tools to reach 30+ total"""
import json
import os

TOOLS_PATH = os.path.join(os.path.dirname(__file__), '..', 'data', 'tools.json')

with open(TOOLS_PATH, 'r') as f:
    tools = json.load(f)

existing_names = {t['name'].lower() for t in tools}
next_id = max(t['id'] for t in tools) + 1

new_tools = [
    # Writing
    {"name": "Sudowrite", "description": "AI creative writing tool designed for fiction authors. Brainstorm plots, develop characters, and refine prose with genre-aware suggestions.", "category": "Writing", "pricing": "Paid", "url": "https://www.sudowrite.com", "needs_vpn": False},
    {"name": "Compose AI", "description": "Free Chrome extension that autocompletes sentences while you type across any website. Saves 40% of typing time.", "category": "Writing", "pricing": "Free", "url": "https://www.compose.ai", "needs_vpn": False},
    {"name": "Cohesive AI", "description": "AI content editor that helps create, refine, and repurpose content across social media, blogs, and ads with brand voice consistency.", "category": "Writing", "pricing": "Freemium", "url": "https://cohesive.so", "needs_vpn": False},
    {"name": "ProWritingAid", "description": "Comprehensive AI writing coach with grammar, style, and readability analysis. In-depth reports for fiction, academic, and business writing.", "category": "Writing", "pricing": "Freemium", "url": "https://prowritingaid.com", "needs_vpn": False},

    # Image
    {"name": "Runway ML", "description": "AI-powered creative suite with Gen-3 video generation, image-to-video, and real-time video editing. Trusted by Hollywood studios.", "category": "Image", "pricing": "Freemium", "url": "https://runwayml.com", "needs_vpn": False},
    {"name": "KREA AI", "description": "Real-time AI image generation canvas with upscaling, style transfer, and custom model training. Drag-and-drop interface for instant results.", "category": "Image", "pricing": "Freemium", "url": "https://www.krea.ai", "needs_vpn": False},
    {"name": "Ideogram", "description": "AI image generator with reliable text rendering in images. Great for logos, posters, and designs requiring accurate typography.", "category": "Image", "pricing": "Freemium", "url": "https://ideogram.ai", "needs_vpn": False},
    {"name": "DreamStudio", "description": "Official Stable Diffusion interface by Stability AI. Full control over generation parameters, negative prompts, and image-to-image editing.", "category": "Image", "pricing": "Paid", "url": "https://beta.dreamstudio.ai", "needs_vpn": False},

    # Video
    {"name": "Animoto", "description": "Drag-and-drop AI video maker for creating professional marketing videos, slideshows, and social media content in minutes.", "category": "Video", "pricing": "Freemium", "url": "https://animoto.com", "needs_vpn": False},
    {"name": "Veed.io", "description": "Browser-based AI video editing suite with auto-subtitles, background removal, screen recording, and one-click translations.", "category": "Video", "pricing": "Freemium", "url": "https://www.veed.io", "needs_vpn": False, "affiliate_link": ""},
    {"name": "Pictory", "description": "AI video editor that automatically extracts highlights from long-form content, adds captions, and creates branded social media clips.", "category": "Video", "pricing": "Freemium", "url": "https://pictory.ai", "needs_vpn": False, "affiliate_link": ""},

    # Audio
    {"name": "Play.ht", "description": "Ultra-realistic AI text-to-speech with 900+ voices across 142 languages. Generate voiceovers, podcasts, and audiobooks with emotion control.", "category": "Audio", "pricing": "Freemium", "url": "https://play.ht", "needs_vpn": False},
    {"name": "Voicemod AI", "description": "Real-time AI voice changer and soundboard for streaming, gaming, and content creation. Create custom voice personas instantly.", "category": "Audio", "pricing": "Freemium", "url": "https://www.voicemod.net", "needs_vpn": False},

    # Code
    {"name": "Codeium", "description": "AI-powered code autocomplete and chat for 70+ languages, 40+ editors. Free for individuals with unlimited completions.", "category": "Code", "pricing": "Free", "url": "https://codeium.com", "needs_vpn": False},
    {"name": "Tabnine", "description": "AI code assistant with whole-line and full-function completions. Trained on permissive open-source code for compliance-friendly suggestions.", "category": "Code", "pricing": "Freemium", "url": "https://www.tabnine.com", "needs_vpn": False},
    {"name": "Mintlify", "description": "AI-powered documentation generator for developers. Auto-generates beautiful API docs, SDK references, and changelogs from code.", "category": "Code", "pricing": "Freemium", "url": "https://mintlify.com", "needs_vpn": False},

    # Productivity
    {"name": "Poe by Quora", "description": "Multi-model AI chat platform giving access to ChatGPT, Claude, Gemini, and more in one unified interface with custom bot creation.", "category": "Productivity", "pricing": "Freemium", "url": "https://poe.com", "needs_vpn": False},
    {"name": "Taskade", "description": "AI-powered workspace combining tasks, mind maps, and team chat. Auto-generate workflows, project plans, and meeting agendas.", "category": "Productivity", "pricing": "Freemium", "url": "https://www.taskade.com", "needs_vpn": False},
    {"name": "Motion AI", "description": "AI calendar and project manager that automatically schedules tasks, meetings, and protects deep work time based on priorities.", "category": "Productivity", "pricing": "Paid", "url": "https://www.usemotion.com", "needs_vpn": False},
    {"name": "Reclaim AI", "description": "Smart calendar assistant that auto-schedules habits, tasks, breaks, and meetings. Defends focus time and syncs across calendars.", "category": "Productivity", "pricing": "Freemium", "url": "https://reclaim.ai", "needs_vpn": False},
    {"name": "Albus AI", "description": "AI knowledge assistant for Slack and web. Ask questions, summarize documents, and extract insights from your team's collective knowledge.", "category": "Productivity", "pricing": "Freemium", "url": "https://www.albus.org", "needs_vpn": False},
    {"name": "Otter.ai", "description": "AI meeting assistant that records, transcribes, and summarizes conversations in real-time. Auto-join, auto-share, and searchable transcripts.", "category": "Productivity", "pricing": "Freemium", "url": "https://otter.ai", "needs_vpn": False},
    {"name": "Fireflies.ai", "description": "AI notetaker for meetings that joins automatically, transcribes, summarizes, and creates searchable knowledge bases from conversations.", "category": "Productivity", "pricing": "Freemium", "url": "https://fireflies.ai", "needs_vpn": False},
    {"name": "Gamma AI", "description": "AI presentation and document creator. Generate beautiful decks, docs, and webpages from prompts with one-click formatting.", "category": "Productivity", "pricing": "Freemium", "url": "https://gamma.app", "needs_vpn": False},
    {"name": "Beautiful.ai", "description": "AI-powered presentation designer with smart templates and auto-layout. Create professional slide decks 10x faster.", "category": "Productivity", "pricing": "Paid", "url": "https://www.beautiful.ai", "needs_vpn": False},
]

filtered = []
for tool in new_tools:
    if tool['name'].lower() not in existing_names:
        filtered.append(tool)
        existing_names.add(tool['name'].lower())
    else:
        print(f"SKIP: {tool['name']}")

added = 0
for i, tool in enumerate(filtered):
    tool['id'] = next_id + i
    tool['rating'] = round(4.0 + (hash(tool['name'] + 'rate') % 100) / 100, 1)
    tool['rating_count'] = (hash(tool['name'] + 'cnt') % 500) + 50
    tool['rating_breakdown'] = {'5': (hash(tool['name']) % 200) + 100, '4': (hash(tool['name'] + 'x') % 150) + 50, '3': (hash(tool['name'] + 'y') % 50) + 10, '2': (hash(tool['name'] + 'z') % 20), '1': (hash(tool['name'] + 'w') % 10)}
    tool['best_for'] = [tool['category'] + ' creators', 'Beginners']
    tool.setdefault('affiliate_link', '')
    tool.setdefault('features', ['AI-powered ' + tool['category'].lower() + ' tool'])
    tools.append(tool)
    added += 1
    print(f"ADDED: #{tool['id']} - {tool['name']} ({tool['category']})")

print(f"\n{added} new tools added. Total: {len(tools)}")

with open(TOOLS_PATH, 'w') as f:
    json.dump(tools, f, indent=2, ensure_ascii=False)

print("Saved.")